### ISSUE TITLE
service discovery consul: CA cert

### ISSUE THREAD

**Issue by chen23** [F5Networks/f5-appsvcs-extension#60]
##### Do you already have an issue opened with F5 support?
No

##### ISSUE TYPE

 - Bug Report

##### AS3 BUILD/ VERSION

```
{
    "version": "3.8.1",
    "release": "1",
    "schemaCurrent": "3.8.0",
    "schemaMinimum": "3.0.0"
}
```

##### BIGIP VERSION

```
Sys::Version
Main Package
  Product     BIG-IP
  Version     13.1.1.2
  Build       0.0.4
  Edition     Point Release 2
  Date        Thu Oct 11 15:32:21 PDT 2018

```

##### SUMMARY

Unable to change CA certificate that is used by Service Discovery.

##### STEPS TO REPRODUCE

```
{
    "class": "ADC",
    "schemaVersion": "3.7.0",
    "id": "urn:uuid:33045210-3ab8-4636-9b2a-c98d22ab425d",
    "label": "Consul Service Discovery",
    "AS3Demo": {
      "class": "Tenant",
      "A1": {
        "class": "Application",
        "template": "http",
        "serviceMain": {
          "class": "Service_HTTP",
          "virtualAddresses": [
            "10.1.10.10"
          ],
          "pool": "web_pool"
        },
        "web_pool": {
          "class": "Pool",
          "monitors": [
            "http"
          ],
          "members": [
            {
              "servicePort": 8080,
              "addressDiscovery": "consul",
              "updateInterval": 10,
              "uri": "https://10.1.20.5:9443/v1/catalog/service/f5-demo-app",
              "encodedToken":"N2NhMTM3MWMtMzFmYS0xMWU4LThiYjItMDIyYzg4ZGFmODRl",
              "credentialUpdate": false
            }
          ]
        }
      }
    }
}
```

##### EXPECTED RESULTS

this should work.

##### ACTUAL RESULTS

```
Fri, 08 Feb 2019 00:29:52 GMT - severe: [appsvcs-discovery] {"message":"Error: 2019-02-08T00:29:50.451Z error:  getNodes.js error code: UNABLE_TO_VERIFY_LEAF_SIGNATURE message: unable to verify the first certificate \n","level":"error"}
Fri, 08 Feb 2019 00:29:52 GMT - severe: [appsvcs-discovery] {"code":500,"error":"2019-02-08T00:29:50.451Z error:  getNodes.js error code: UNABLE_TO_VERIFY_LEAF_SIGNATURE message: unable to verify the first certificate \n","level":"error"}
Fri, 08 Feb 2019 00:29:53 GMT - severe: [appsvcs] {"message":"Error: POST http://admin:XXXXXX@localhost:8100/mgmt/tm/cli/script run cli script response=400 body={\"code\":400,\"message\":\"script did not successfully complete, status:1\",\"errorStack\":[],\"apiError\":26214401}","code":"500","level":"error"}
```
I also tried updating /config/ssl/ssl.crt/ca-bundle.crt w/ the same results.

**Comment by chen23** [comment]
Looking at this further

Based on https://github.com/F5Networks/f5-cloud-libs-consul/blob/develop/lib/consulCloudProvider.js#L124 

I suspect that one may need to specify “ca-bundle.crt” as a parameter to whatever function is ultimately performing the HTTP request.  Specifically to set an options of 

```
options = {ca:fs.readFileSync(‘/config/ssl/ssl.crt/ca-bundle.crt’) }
```

The other option would be to create a local virtual server that proxies requests to the backend consul server (i.e. http://172.16.10.11 -> https://10.1.20.5:9443 and does any certificate verification / client-side certificate).

**Comment by kaustriaf5** [comment]
@chen23 thanks for reporting, this is being tracked internally as 887.

**Comment by dstokesf5** [comment]
Looks like this was resolved in version 3.11, or possible 3.12. Closing this issue. If you are still experiencing a problem, please reopen this ticket or create a new one.
